src/main/java/com/kodar/academy/Library/service/impl/BookServiceImpl.java

```diff
@@ -69,4 +84,27 @@ public BookResponseDTO addBook(BookCreateDTO bookCreateDTO) {
         return BookMapper.mapToResponse(book);
     }
 
+    @Override
+    @Transactional
+    public BookResponseDTO editBook(int id, BookEditRequestDTO bookEditRequestDTO) {
+        Book oldBook = bookRepository.findById(id).orElseThrow();
+        String oldTitle = oldBook.getTitle();
+        String oldPublisher = oldBook.getPublisher();
+        oldBook.setTitle(bookEditRequestDTO.getTitle());
+        oldBook.setPublisher(bookEditRequestDTO.getPublisher());
```

Comment:
You need the old/new value only if they are different. Also execute setX only if the old/new value are different. Move this logic inside the if statements.